src/audio/host-zephyr.c

```diff
@@ -931,6 +919,30 @@ static int host_params(struct comp_dev *dev,
 	return err;
 }
 
+/* configure the DMA params and descriptors for host buffer IO */
+static int host_params(struct comp_dev *dev,
+		       struct sof_ipc_stream_params *params)
+{
+	struct host_data *hd = comp_get_drvdata(dev);
+	int err;
+
+	comp_dbg(dev, "host_params()");
+
+	err = host_verify_params(dev, params);
+	if (err < 0) {
+		comp_err(dev, "host_params(): pcm params verification failed.");
+		return -EINVAL;
+	}
+
+	err = host_params_dma(hd, dev, params, &dev->bsink_list, &dev->bsource_list,
+			      dev->pipeline, dev->frames, dev == dev->pipeline->sched_comp);
```

Comment:
everywhere you call `host_params_dma()` parameters 4-7 are the same: `&dev->bsink_list, &dev->bsource_list, dev->pipeline, dev->frames, dev == dev->pipeline->sched_comp`, can we just pass `dev` a second time instead, because in one case it's different from `dev` in the second parameter?

Reply:
actually, they are different, please check copier_params calling:

					ret = host_params_dma(cd->hd, dev, params,
							      &child_dev->bsink_list,
							      &child_dev->bsource_list,
							      child_dev->pipeline,
							      child_dev->frames,
							      is_scheduling_source);

it is 2 different devices, and child_dev are designed for remove, will update more later.

Reply:
@btian1 that's exactly what I meant, sorry for being unclear. So you could have
```
ret = host_params_dma(cd->hd, dev, params, child_dev);
```
and for other locations like the one above at line 937 you'd have
```
err = host_params_dma(hd, dev, params, dev);
```
and inside the function you'll have your
```
int host_params_dma(xxx *hd, struct comp_dev *dev, params, struct comp_dev *dev2)
{
	struct list_item *sink_list = &dev2->bsink_list;
	struct list_item *source_list = &dev2->bsource_list;
	...
	bool is_scheduling_source = dev2 == dev2->pipeline->sched_comp;
}
```
but please replace `dev2` with some meaningful name